projects/JohannesMerkt/proposal/readme.md

```diff
@@ -0,0 +1,30 @@
+# Proposal: Draw Geometry with Code
+
+Johannes Merkt
+
+31.12.2021
+
+## Abstract
+
+Shaders can be used to draw images with code. At shadertoy.com you can find many astonishing examples. This project wants to move this concept to the 3d world with geometry, allowing a user to write a shader to draw 3d geometry.
+
+## Motivation
+
+This project will take in a shader that describes a 3D volume with code and turns this into a geometry by analizing the density of the volume. It might also be possible to describe this volume with color too and turn this color information into vertex color of the geometry. In order to make this possible I would have to write a meshing algorithm that takes in the shader and generates a geometry as fast as possible.
+
+Drawing with shaders is very powerful, it allows for high performant art that updates in real time, is procedural and can also be interactive. I find that fascinating. But what if I could write shaders that describe 3d Geometry? Then you could create complex geometry in a completly new way which is also interesting for others. I hope I will be able to make this proposed system performant enough to draw geometry in real time. If not it is still useful but has to be rendered or cached first for real time viewing.
```

Comment:
> I hope I will be able to make this proposed system performant enough to draw geometry in real time.

Based on the description of this section, I am currently lack the understanding of the theoretical basis of implementing this idea.